packages/taiko-client/preconfapi/builder/interface.go

```diff
@@ -2,18 +2,54 @@ package builder
 
 import (
 	"context"
+	"encoding/hex"
+	"strings"
 
-	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/ethereum/go-ethereum/rlp"
 )
 
+type BuildUnsignedOpts struct {
+	BlockOpts []BlockOpts
+}
+type BlockOpts struct {
+	L1StateBlockNumber uint32
+	Timestamp          uint64
+	SignedTransactions []string
+	Coinbase           string
+	ExtraData          string
+}
 type TxBuilder interface {
 	BuildUnsigned(
 		ctx context.Context,
-		txListBytes []byte,
-		l1StateBlockNumber uint32,
-		timestamp uint64,
-		coinbase common.Address,
-		extraData [32]byte,
+		opts BuildUnsignedOpts,
 	) (*types.Transaction, error)
 }
+
+func signedTransactionsToTxListBytes(txs []string) ([]byte, error) {
+	var transactions types.Transactions
+
+	for _, signedTxHex := range txs {
+		signedTxHex = strings.TrimPrefix(signedTxHex, "0x")
```

Comment:
Use `hexutil.Decode`, then don't need to trim prefix `0x`.

Reply:
will do!

Reply:
the user is passing these in, we don't know if they have 0x or not

Reply:
`hexutil.Decode` function will do this kind of inspection work.